django/db/models/sql/query.py

```diff
@@ -65,20 +65,58 @@ def get_field_names_from_opts(opts):
     )
 
 
+def get_paths_from_expression(expr):
+    if isinstance(expr, F):
+        yield expr.name
+    elif hasattr(expr, "flatten"):
+        for child in expr.flatten():
+            if isinstance(child, F):
+                yield child.name
+            elif isinstance(child, Q):
+                yield from get_children_from_q(child)
+
+
 def get_children_from_q(q):
     for child in q.children:
         if isinstance(child, Node):
             yield from get_children_from_q(child)
-        else:
-            yield child
+        elif isinstance(child, tuple):
+            lhs, rhs = child
+            yield lhs
+            if hasattr(rhs, "resolve_expression"):
+                yield from get_paths_from_expression(rhs)
+        elif hasattr(child, "resolve_expression"):
+            yield from get_paths_from_expression(child)
 
 
-def get_child_with_renamed_prefix(prefix, replacement, child):
+def replace_child_name(child_name, prefix, replacement):
+    return child_name.replace(prefix, replacement, 1)
```

Comment:
It's one-liner, IMO, it's not worth extra function calls. I'd move this logic directly to the `get_child_with_renamed_prefix()`.